client/js/table-view.js

```diff
@@ -86,26 +165,229 @@ class TableView {
 		this.sheetBodyEl.appendChild(fragment);
 	}
 
+	isCurrentCell(col, row) {
+		return this.currentCellLocation.col === col &&
+		       this.currentCellLocation.row === row;
+	}
+
 	attachEventHandlers() {
+
+		const context = this;
+
 		this.sheetBodyEl.addEventListener('click', this.
 			handleSheetClick.bind(this));
 		this.formulaBarEl.addEventListener('keyup', this.
 			handleFormulaBarChange.bind(this));
+		
+		this.addRowEl.addEventListener('click', this.
+			handleAddRowClick.bind(this));
+		this.addColumnEl.addEventListener('click', this.
+			handleAddColumnClick.bind(this));
+
+		this.rowLabelsEl.addEventListener('click', this.
+			handleRowLabelClick.bind(this));
+
+		this.headerRowEl.addEventListener('click', this.
+			handleColHeaderClick.bind(this));
+
+		// this works fine
+		this.computeSumFormulaEl.addEventListener('click', this.
+			handleFormulaBarEnter.bind(this));
+
+		// this doesn't work...
+		/*this.formulaBarEl.addEventListener('keypress', function(e) {
+			if (e.key === 'Enter') {
+				console.log('hi');
+				console.log(context.handleFormulaBarEnter);
+				console.log(context)
+				context.handleFormulaBarEnter.bind(e);
+			}
+		});*/
+			
+	}
+
+	handleFormulaBarEnter(event) {
+		const value = this.formulaBarEl.value;
+		
+		// to potentially use sum formula...
+		// must start with correct command, include colon, end with closing paren
+		if (value.substring(0, 5) === "=SUM(" && 
```

Comment:
please move this "formula detection" out into a helper function, probably in its own module. Maybe call it `formula-helper.js` or some such?